s4/clarity/lims.py

```diff
@@ -91,61 +92,102 @@ def __init__(self, root_uri, username, password, dry_run=False, insecure=False,
+        self.reagent_lots = ElementFactory(self, ReagentLot)
 
-        self.reagent_kits = ElementFactory(self, ReagentKit, batch_flags=BatchFlags.QUERY)
+        self.reagent_kits = ElementFactory(self, ReagentKit)
 
-        self.reagent_types = ElementFactory(self, ReagentType, batch_flags=BatchFlags.QUERY)
+        self.reagent_types = ElementFactory(self, ReagentType)
 
-        self.researchers = ElementFactory(self, Researcher, batch_flags=BatchFlags.QUERY)
+        self.researchers = ElementFactory(self, Researcher)
 
-        self.labs = ElementFactory(self, Lab, batch_flags=BatchFlags.QUERY)
+        self.labs = ElementFactory(self, Lab)
 
-        self.roles = ElementFactory(self, Role, batch_flags=BatchFlags.QUERY)
+        self.roles = ElementFactory(self, Role)
 
-        self.permissions = ElementFactory(self, Permission, batch_flags=BatchFlags.QUERY)
+        self.permissions = ElementFactory(self, Permission)
 
         # configuration
         from .configuration import Workflow, Protocol, ProcessType, Udf, ProcessTemplate, Automation
 
-        self.workflows = ElementFactory(self, Workflow, batch_flags=BatchFlags.QUERY,
-                                        request_path='/configuration/workflows')
-        self.protocols = ElementFactory(self, Protocol, batch_flags=BatchFlags.QUERY,
-                                        request_path='/configuration/protocols')
-        self.udfs = UdfFactory(self, Udf, batch_flags=BatchFlags.QUERY,
-                               request_path='/configuration/udfs')
-        self.process_types = ElementFactory(self, ProcessType, batch_flags=BatchFlags.QUERY,
-                                            name_attribute="displayname")
-        self.process_templates = ElementFactory(self, ProcessTemplate, batch_flags=BatchFlags.QUERY,
-                                                name_attribute="name")
-        self.automations = ElementFactory(self, Automation, batch_flags=BatchFlags.QUERY,
-                                          name_attribute="name", request_path="/configuration/automations")
+        self.workflows = ElementFactory(self, Workflow)
+
+        self.protocols = ElementFactory(self, Protocol)
+
+        self.udfs = UdfFactory(self, Udf)
+
+        self.process_types = ElementFactory(self, ProcessType)
+
+        self.process_templates = ElementFactory(self, ProcessTemplate)
+
+        self.automations = ElementFactory(self, Automation)
 
         self.stages = ElementFactory(self, Stage)
 
+    def _get_hostname(self):
+        # type: () -> str
+        """
+        Verify that we have a valid hostname in our root uri
+        and isolate the host name from it.
+        :return: The name of the Clarity server.
+        """
+
+        # Compile a regex statement to isolate the host name
+        # and verify that our root uri looks like a properly formed URI
+        hostname_finding_regex = re.compile(r'https?://([^/:]+)')
```

Comment:
It's inefficient to compile a regex every time you instantiate a new LIMS object, which is why this was an internal-use-only class constant instead.

The difference is probably so microscopic it probably doesn't really matter, but this seems like an unnecessary change...

Reply:
We almost never create more than one LIMS object, and if we were to create more than one the compile time would be unnoticeable in the full init process.
What it does get us is the actual regex statement in the place that it is used, so that if/when someone comes to this code to read it they can see the full logic involved. I am treating it as a piece of code, kept with the function in which it is executed, as opposed to defining it as a constant at the top of the file.